authorizer.go

```diff
@@ -158,8 +159,9 @@ func (v *authorizer) Authorize() error {
 			break
 		}
 		for _, query := range policy.Queries {
-			res := v.world.QueryRule(query.convert(v.symbols), v.symbols)
-			if len(*res) != 0 {
+			// TODO: how to deal with err? len(res) can be 0 for checks with pred that don't have Vars
```

Comment:
`len(res)` will not be 0 if the check succeed, even without vars. If we have a rule like `fact(0) <- 1 < 2` or `fact(0) <- a(1), b(2)`, `fact(0)` will be generated if it matches and `len(res) == 1`

Reply:
so for all checks and policies the condition should be `len(*res) != 0 && err != nil`